src/abio/node/io.cljs

```diff
@@ -1,59 +1,50 @@
 (ns abio.node.io
   (:require
-    [abio.io :as io]
-    [clojure.string :as string]))
+   [abio.io :as io]
+   [clojure.string :as string]))
 
-(defrecord BufferedReader [raw-read raw-close buffer pos]
+(defrecord BufferedReader [path opts fs] ; `opts` should contain :encoding and optionally :flag
```

Comment:
Do we want to pass the fs module around? Usually it can be required once like a namespace?

Reply:
I mostly did that because I don't like to bake in stuff like that, mostly in the off-chance I want to reuse it later.

That being said, it's probably unrealistic to think that way about the host specific implementations...

Reply:
In general I agree with you and it is always though to strike the balance